src/node_buffer.cc

```diff
@@ -1526,6 +1526,14 @@ uint32_t FastWriteString(Local<Value> receiver,
                          uint32_t max_length,
                          // NOLINTNEXTLINE(runtime/references)
                          FastApiCallbackOptions& options) {
+  // Just a heads up... this is a v8 fast api function. The use of
+  // FastOneByteString has some caveats. Specifically, a GC occurring
+  // between the time the FastOneByteString is created and the time
+  // we use it below can cause the FastOneByteString to become invalid
+  // and produce garbage data. This is not a problem here because we
+  // are not performing any allocations or other operations that would
+  // trigger a GC before the FastOneByteString is used. Take care when
+  // modifying this code to ensure that no operations would trigger a GC.
```

Comment:
Can we do something like `DebugSealHandleScope`, i.e. actually enforce this constraint through code? E.g. a `DebugDisallowJavascriptExecutionScope` that defers to V8's `DisallowJavascriptExecutionScope`? (That may not be 100% equivalent to disallowing GC, but maybe it's close enough?)

Reply:
@jasnell Not a blocker by any means, it just seems simpler that way